codelabs/background_segmenter/code/index.js

```diff
@@ -0,0 +1,342 @@
+// we start by importing the tasks-vision module from the CDN
+import vision from "https://cdn.skypack.dev/@mediapipe/tasks-vision";
+
+// the Camera class will help us to interact with the webcam
+import Camera from "./camera.js";
+import {downloadImage, scaleImageData} from "./utils.js";
+const { ImageSegmenter, SegmentationMask, FilesetResolver } = vision;
+
+// get a reference to the DOM elements that we wil use
+const btnStart = document.getElementById('btnStart');
+const btnStop = document.getElementById('btnStop');
+const video = document.getElementById('video');
+const selVideoSource = document.getElementById('selectVideoSource');
+const selVideoResolution = document.getElementById('selectVideoResolution');
+const txtBackgroundImageInput = document.getElementById('txtBackgroundImage');
+
+// get a reference to the canvas element and its context
+const videCanvas = document.getElementById('canvas');
+const videoCanvasCtx = videCanvas.getContext('2d');
+
+// create a new camera instance
+const camera = new Camera(video, video.videoWidth, video.videoHeight);
+
+// initialize variables that will be used later
+let backgroundImage = null;
+let requestAnimationFrameId = null;
+let imageSegmenter = null;
+let labelsToSegment = ["dog", "person"]
+const labels = [
+    'background',
+    'aeroplane',
+    'bicycle',
+    'bird',
+    'boat',
+    'bottle',
+    'bus',
+    'car',
+    'cat',
+    'chair',
+    'cow',
+    'dining table',
+    'dog',
+    'horse',
+    'motorbike',
+    'person',
+    'potted plant',
+    'sheep',
+    'sofa',
+    'train',
+    'tv'];
+
+// this function will be called when the webpage is loaded
+document.addEventListener('DOMContentLoaded', async () =>{
+    // populate the video source select with the available video sources
+    await populateVideoSourceSelect();
+    // create a new image segmenter instance using the mediapipe wasm runtime
+    imageSegmenter = await createImageSegmenter();
+    // initialize the materialize components
+    M.AutoInit();
+});
+
+// segmentation stuff
+async function createImageSegmenter() {
+    /**
+     * Creates a new image segmenter
+     * @returns {Promise<ImageSegmenter>} imageSegmenter
+     */
+    const wasmFileset = await FilesetResolver.forVisionTasks(
+        "https://cdn.jsdelivr.net/npm/@mediapipe/tasks-vision@latest/wasm"
+    );
+    // Initializes the Wasm runtime and creates a new image segmenter from the provided options.
+    return await  ImageSegmenter.createFromOptions(wasmFileset, {
+        baseOptions: {
+            modelAssetPath:
+                "https://storage.googleapis.com/mediapipe-assets/deeplabv3.tflite?generation=1661875711618421"
+
+        },
+        runningMode: "VIDEO",
```

Comment:
add `outputType: "CONFIDENCE_MASK"` since this is more of a selfie demo than anything else, there's no need to identify what type of object you're masking. And confidence mask gives you the ability to blend pixels for a cleaner edge